Source/SPTableContent.h

```diff
@@ -242,6 +242,8 @@ typedef NS_ENUM(NSInteger, SPTableContentFilterSource) {
 - (void)clickLinkArrowTask:(SPTextAndLinkCell *)theArrowCell;
 - (void)updateResultStore:(SPMySQLStreamingResultStore *)theResultStore approximateRowCount:(NSUInteger)targetRowCount;
 - (BOOL)saveRowToTable;
+- (BOOL)_saveRowToTable;
```

Comment:
not happy about this method being the same name as the one above just with underscore... what's the difference? is there docu in header file?

Reply:
Also, can't seem to find this method - is it `- (BOOL)_saveRowToTableWithQuery:(NSString*)queryString{`?

Reply:
let me check .. i renamed some things

Reply:
Yes, missed the rename in the header. Actually moved it to the m file as it should only be called within the class.

`saveRowToTable` originally did two things:

1. Figure out what had changed so it could construct an SQL query.
2. Executed the query.

We wanted to put an optional check before executing the query, so I split the method up:

`deriveQueryString` does part 1.

`saveRowToTable` calls  `deriveQueryString` so I can display the query string in the Alert. 

If the user clicks cancel, do nothing, just reload the table to restore and changed cells.
If the user clicks Proceed, call `_saveRowToTableWithQuery` with the query string from `deriveQueryString`

I'll add this to the docs.